src/test/java/com/dpgil/pathlinker/path_linker/internal/PathLinkerModelTest.java

```diff
@@ -0,0 +1,594 @@
+		originalNetworkMap = new HashMap<String, Long>();
+		originalNetworkDir = support.getNetwork();
+		originalNetworkDir.getDefaultEdgeTable().createColumn("edge_weight", Double.class, false);
+
+		//access input file
+		url = PathLinkerModelTest.class.getResource("/input/graph-dir_human-interactome.txt");
+		file = new File(url.getFile());
+
+		//construct network based on input file
+		@SuppressWarnings("resource")
+		BufferedReader readerInput = new BufferedReader(new FileReader(file));
+
+		for (String line = readerInput.readLine(); line != null; line = readerInput.readLine()) {
+			String[] arr = line.split("\\s+");
+			CyNode source, target;
+
+			//to avoid adding same source
+			if (!originalNetworkMap.containsKey(arr[0])) {
+				source = originalNetworkDir.addNode();
+				originalNetworkDir.getRow(source).set(CyNetwork.NAME, arr[0]);
+				originalNetworkMap.put(arr[0], source.getSUID());
+			} else source = originalNetworkDir.getNode(originalNetworkMap.get(arr[0]));
+
+			//to avoid adding same target
+			if (!originalNetworkMap.containsKey(arr[1])) {
+				target = originalNetworkDir.addNode();
+				originalNetworkDir.getRow(target).set(CyNetwork.NAME, arr[1]);
+				originalNetworkMap.put(arr[1], target.getSUID());
+			} else target = originalNetworkDir.getNode(originalNetworkMap.get(arr[1]));
+
+			CyEdge edge = originalNetworkDir.addEdge(source, target, true); //set true because the the graph is directed
+			originalNetworkDir.getRow(edge).set("edge_weight", Double.parseDouble(arr[2]));
+		}
+
+		//================================= setting up undirect network =====================================
+		originalNetworkMap = new HashMap<String, Long>();
+		originalNetworkUndir = support.getNetwork();
+		originalNetworkUndir.getDefaultEdgeTable().createColumn("edge_weight", Double.class, false);
+
+		//access input file
+		url = PathLinkerModelTest.class.getResource("/input/graph-undir_human-interactome.txt");
+		file = new File(url.getFile());
+
+		//construct network based on input file
+		readerInput = new BufferedReader(new FileReader(file));
+
+		for (String line = readerInput.readLine(); line != null; line = readerInput.readLine()) {
+			String[] arr = line.split("\\s+");
+			CyNode source, target;
+
+			//to avoid adding same source
+			if (!originalNetworkMap.containsKey(arr[0])) {
+				source = originalNetworkUndir.addNode();
+				originalNetworkUndir.getRow(source).set(CyNetwork.NAME, arr[0]);
+				originalNetworkMap.put(arr[0], source.getSUID());
+			} else source = originalNetworkUndir.getNode(originalNetworkMap.get(arr[0]));
+
+			//to avoid adding same target
+			if (!originalNetworkMap.containsKey(arr[1])) {
+				target = originalNetworkUndir.addNode();
+				originalNetworkUndir.getRow(target).set(CyNetwork.NAME, arr[1]);
+				originalNetworkMap.put(arr[1], target.getSUID());
+			} else target = originalNetworkUndir.getNode(originalNetworkMap.get(arr[1]));
+
+			CyEdge edge = originalNetworkUndir.addEdge(source, target, true);
```

Comment:
This is not an undirected network. To add an undirected edge, you need to set the "directed" (third) option to false. It would probably be more clear if you made a boolean variable for directed/undirected.